carbons/carbons.go

```diff
@@ -87,3 +88,24 @@ func WrapSent(delay delay.Delay, r xml.TokenReader) xml.TokenReader {
 		xml.StartElement{Name: xml.Name{Space: NS, Local: "sent"}},
 	)
 }
+
+// Unwrap unwraps a carbon copied message and returns
+// a stream corresponding to the message.
+// A bool is returned and is set to true if the type of the carbon-copy
+// is "sent", otherwise it indicated that it's "received".
+func Unwrap(del *delay.Delay, r xml.TokenReader) (xml.TokenReader, bool, error) {
```

Comment:
I'm not sure about the `bool` being returned; if you're reading the function signature it's not obvious what it would mean and even reading the docs it's a little confusing. I'm not sure what would be better though: maybe just returning the `xml.StartElement` and letting the user decide what to do with it? Or maybe a new type with constants for Sent/Received:

```go
type Direction bool

const (
  Sent        = true
  Received = false
)
```

I'm leaning towards the startelement, but maybe I'm overthinking it and the way it is now is fine too?

Reply:
I based my choice on the signature of the handler in [here](https://github.com/mellium/xmpp/blob/main/carbons/handler.go#L31) but I share the same feelings about the readability. I think we can go with the `xml.StartElement` option.

Reply:
Oh yah, good point, I forgot about that one. Oh well, that was a mistake on my part.